equipment/equipment.lua

```diff
@@ -9,7 +10,70 @@ ftype.base.fields.item.type.fields.item = {
     end,
 }
 
-return data
+ftype.base.fields.equip = {
+    data = function(equipment_slot, item)
+        local bag = item.bag
+        local index = item.index
+        assert(resources.bags[bag].equippable, "Can not equip from this bag (bag = " .. bag .. ")")
+        assert(item.id ~= 0, "Can not equip from an empty bag slot (bag = " .. bag .. ", index = " .. index .. ")")
+
+        packets.outgoing[0x050]:inject({bag_index = index, slot_id = equipment_slot.slot, bag_id = bag})
+    end,
+}
+
+ftype.base.fields.unequip = {
+    data = function(equipment_slot)
+        packets.outgoing[0x050]:inject({bag_index = 0, slot_id = equipment_slot.slot, bag_id = 0})
+    end,
+}
+
+local equipment = {}
+
+equipment.equip = function(slot_items)
```

Comment:
I'd change the signature to `function(_, slot_items)`, just so you call it with `:equip`, similar to other functions. We're trying to go for this style globally, so we'd like to keep the consistency.